src/processing.py

```diff
@@ -0,0 +1,280 @@
+import sys
+import os
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+import polars as pl
+import os
+import numpy as np
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy import create_engine, Column, Integer, String, Float, DateTime
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import inspect
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.declarative import DeclarativeMeta
+
+def load_data(file_path) -> pl.DataFrame:
+    try:
+        if isinstance(file_path, str):
+            csv_format = file_path.endswith(".csv")
+            excel_format = file_path.endswith(".xlsx")
+            
+            #if the file exists in the working directory
+            if csv_format:
+                data_read_csv = pl.read_csv(file_path)
+                if data_read_csv.is_empty() == False:
+                    return data_read_csv
+                else:
+                    return "Error reading csv file"
+            elif excel_format:
+                data_read_excel = pl.read_excel(file_path)
+                if data_read_excel.is_empty() == False:
+                    return data_read_excel
+                else:
+                    return "Error reading excel file"
+            else:
+                return "File path does not exist or is not in the right format"
+        else:
+            return "Invalid file path or format provided"
+
+    except ValueError as e:
+        print(f"An error occurred: {e}")
+        return e
+    
+
+
+def check_missing_duplcates(df: pl.DataFrame, df_name: str):
+    null_df = df.filter(df.is_empty() == True)
+    null_count = null_df.shape[0]
+    print(f"{df_name} Null count: {null_count}")
+    unique_count = df.n_unique()
+    filtered_df = df.filter(df.is_duplicated() == False)
+    if unique_count == filtered_df.shape[0]:
+        print(f"No duplicates found in {df_name}")
+
+
+#i Want to link these tables together and establish a fact table
+#I will join the dataframes on the common columns
+#I wnant to create a unique for the product category dataframe,
+#  so that we can join it with the other dataframes
+
+def transform_product_category_df(df) -> pl.DataFrame:
```

Comment:
Add type hints to all functions in this PR

Reply:
Noted